core/navigation/src/main/kotlin/nl/q42/template/navigation/viewmodel/NavigationState.kt

```diff
@@ -3,8 +3,15 @@ package nl.q42.template.navigation.viewmodel
 import java.util.UUID
 
 sealed class NavigationState {
-    object Idle : NavigationState()
-    data class NavigateToRoute(val route: String, val id: String = UUID.randomUUID().toString()) : NavigationState()
+    data object Idle : NavigationState()
+    data class NavigateToRoute(
+        val route: String,
+        val addToBackStack: Boolean,
+        val clearBackStack: Boolean,
```

Comment:
"Clear back stack" is clear 😜  but "Add to back stack" is ambiguous to me when it's false:
- if **addToBackStack is false** and clearBackStack is true, then the literal meaning is to _not add that screen to any back stack_, when in reality I assume it will clear the back stack **and then add it** to the back stack as the first element
- if **addToBackStack is false** and clearBackStack is false, then the literal meaning is to _not add that screen to any back stack_, when in reality I assume it will replace the current entry in the back stack
- when clearBackStack is true, is there a difference between setting addToBackStack to true or false?

With my concerns in mind, do you have ideas on how to improve the API to avoid ambiguity and possible illegal/redundant combinations of flags?

My suggestion is that if some states can't be combined, it's better to use an enum or sealed interface to represent them.

Reply:
See the google navigation lib's docs, they use the same term: `By calling addToBackStack(), the replace transaction is saved to the back stack so the user can reverse the transaction and bring back the previous fragment by pressing the Back button.`

it's the same here, you can think of it as "the transaction".

I can make it a sealed class.

Reply:
That's a function and this PR is adding a boolean which in terms of API is not the same. On the other hand, I personally don't regard any of the navigation libraries from Google as great APIs 😉 so I would probably prefer an API from Frank in this case